assets/js/diff_viewer_component.js

```diff
@@ -8,19 +8,39 @@ const DiffViewerComponent = {
     this._renderDiff();
   },
   _renderDiff() {
+    const sourceUrl = this.el.getAttribute("data-target-github-url");
     const diff2htmlUi = new Diff2HtmlUI(this.el, this.el.getAttribute("data-diff"),{
         drawFileList: false,
         outputFormat: this.el.getAttribute("data-view-type"),
         highlight: true,
         fileContentToggle: false,
-        matching: 'words'
+        matching: 'words',
+        rawTemplates: {'generic-file-path': this._genericFilePathTemplate(sourceUrl)}
     });
 
     diff2htmlUi.draw();
 
     const event = new CustomEvent("phx-diff:diff-loading-stop", {bubbles: true, cancelable: true});
     window.dispatchEvent(event);
   },
+  // https://github.com/rtfpessoa/diff2html/blob/master/src/templates/generic-file-path.mustache
+  _genericFilePathTemplate(sourceUrl) {
+    return `
+        <span class="d2h-file-name-wrapper">
+          {{>fileIcon}}
+          <a class="hover:underline" href="${sourceUrl}/{{fileDiffName}}">
```

Comment:
I was looking at this [Github diff](https://github.com/phoenix-diff/phoenix-diff/commit/304d9bdceb45821a98ab733e788b6db92af3159f#diff-28da1bdd5726054b085c36e74a9694420c1b8313c1f61fe7298d22b530d7b655) and it goes to the target url unless the file was deleted and then it goes to the source url. Could that functionality be available here?

Reply:
i looked a bit into `diff2Html` and unfortunately the template doesn't know if the file was deleted or not. We'd have to put the links in the e.g. `tag-file-*` templates to point to source or target depending on if the file was deleted or not: https://github.com/rtfpessoa/diff2html/tree/c7248f03d7b287ff269ae902f6d63fe5a57d031a/src/templates

Reply:
i've added links to the source + target files to all diffs for now

Reply:
Unfortunately this doesn't work for file renames, since the `fileDiffName` will be `assets/css/{app.scss → app.css}` and diff2Html doesn't expose any other variables we could be using in this template :(
https://github.com/rtfpessoa/diff2html/blob/master/src/line-by-line-renderer.ts#L75

Reply:
Sorry for changing my mind, but I liked UI you had previously where the filename was the link to github. Let's just pick the url from the file for the target app spec for now.